src/components/Collapse.js

```diff
@@ -1,21 +1,31 @@
-import { useState } from 'react'
+import { useCallback, useEffect, useState } from 'react'
 import { div, h } from 'react-hyperscript-helpers'
 import { Link } from 'src/components/common'
 import { icon } from 'src/components/icons'
 
 
-const Collapse = ({ title, buttonStyle, initialOpenState, children, titleFirst, ...props }) => {
+const Collapse = ({ title, buttonStyle, initialOpenState, children, titleFirst, onFirstOpen = () => {}, refreshKey, ...props }) => {
   const [isOpened, setIsOpened] = useState(initialOpenState)
   const angleIcon = icon(isOpened ? 'angle-down' : 'angle-right', { style: { marginRight: '0.25rem', flexShrink: 0 } })
 
+  const firstOpenCallback = useCallback(onFirstOpen, [refreshKey])
+
+  useEffect(() => {
+    if (isOpened) {
+      firstOpenCallback()
+    }
+  }, [firstOpenCallback, isOpened, refreshKey])
```

Comment:
`useEffect` seems to be triggered every time I click on a snapshot table in the left nav. This means that `firstOpenCallback()` is called on every click, and thus the API call to `.../entities?...` occurs on every click.

Found by opening dev tools, watching network requests,  and clicking around on the tables.